# HG changeset patch
# Date 1627549531 -10800
#      Thu Jul 29 12:05:31 2021 +0300
# Branch conda
# Node ID 49ba58431c069c0f944f12f935c564d538feee79
# Parent  729ac336ae0c7408d4df2f31854e37c2aeb65199
adapt regex to sys.version, add parsing compiler too

diff -r 729ac336ae0c -r 49ba58431c06 lib-python/3/platform.py
--- a/lib-python/3/platform.py
+++ b/lib-python/3/platform.py
@@ -1140,9 +1140,9 @@
 )
 
 _pypy_sys_version_parser = re.compile(
-    r'([\w.+]+)\s*'
+    r'([\w.+]+)[^(]*'
     r'\(#?([^,]+),\s*([\w ]+),\s*([\w :]+)\)\s*'
-    r'\[PyPy [^\]]+\]?')
+    r'\[PyPy [^\]]+with ([^\]]*)\]?')
 
 _sys_version_cache = {}
 
@@ -1213,8 +1213,7 @@
         if match is None:
             raise ValueError("failed to parse PyPy sys.version: %s" %
                              repr(sys_version))
-        version, buildno, builddate, buildtime = match.groups()
-        compiler = ""
+        version, buildno, builddate, buildtime, compiler = match.groups()
 
     else:
         # CPython
@@ -1246,7 +1245,7 @@
         version = '.'.join(l)
 
     # Build and cache the result
-    result = (name, version, branch, revision, buildno, builddate, compiler)
+    result = (name, version, branch, revision[:8], buildno, builddate, compiler)
     _sys_version_cache[sys_version] = result
     return result
 
